redhat-appstudio/service-provider-integration-operator
# Flip token binding to awaiting state if data suddenly disappears & clean sync objects

### What does this PR do?
Flips token binding to awaiting state if token data suddenly disappears from storage & clean synced objects (secrets & sa)

### Screenshot/screencast of this PR



### What issues does this PR fix or reference?

https://issues.redhat.com/browse/SVPI-414

### How to test this PR?

 
-  Create binding, upload token.
-  Delete  token data from storage
-  Binding must appear in the `AwaitingTokenData` state

```
quay.io/redhat-appstudio/pull-request-builds:spi-controller-#
quay.io/redhat-appstudio/pull-request-builds:spi-oauth-#
```
